api/config/dex/config_request.go

```diff
@@ -169,6 +167,15 @@ func (c *ConfigRequest) PrepareSystemConfig(creds *shared.TLSCredentials) (share
 	// default name_id_policy_format (SAML)
 	c.V1.Sys.GetConnectors().GetSaml().setNameIDPolicyDefault()
 
+	if c.V1.Sys.GetExpiry().GetIdTokens() == nil {
+		// Different defaults when SAML is or isn't used
+		if c.V1.Sys.GetConnectors().GetSaml() != nil {
+			c.V1.Sys.Expiry.IdTokens = w.String("24h")
+		} else {
+			c.V1.Sys.Expiry.IdTokens = w.String("3m")
+		}
```

Comment:
Is this selecting 24 hours when the active connector is SAML or just when SAML is one of the possible login choices? With the name `GetConnectors().GetSaml()` it feels like that is just checking the latter, but that is not the behavior we want, right? (I tried to find the source of that code in VSCode but F12 was not being my friend.)

Reply:
Re-posting what I had shared in slack for visibility:

👉If Saml is among the connectors, we need the large default. Token expiry is not per connector, unfortunately.

Also

> I tried to find the source of that code in VSCode but F12 was not being my friend.

I don't know how the indexing works, but the `GetSaml()` etc helpers are code generated from the protobuf definitions.